packages/extension/src/RichTextActions/RichTextActionsImpl.tsx

```diff
@@ -0,0 +1,385 @@
+  - Do not include images in your response.
+
+If the user's prompt asks for only a portion of a document, you should return only that portion.
+  - If the user asks for only a title, produce only a title.
+  - If the user asks for only a paragraph, produce only a paragraph.
+  - If the user asks for only a bullet list, produce only a bullet list.
+
+If the user provides a prompt that you cannot produce a document for, you should return an error.
+  - Prefix any errors with [DIALOG].
+  - For example, if the user's prompt is \`Hello?\`, then you might respond with \`[DIALOG] Please describe the content you want to generate.\`
+
+Do not converse with the user.
+  - Do not ask clarifying questions.
+  - Do not add conversational preamble such as \`Sure, I can do that\`, or \`Ok, here's the change\`.
+  - Do not include [DIALOG] if you've successfully produced a document.`
+        },
+        {
+          role: "user",
+          content: prompt
+        }
+      ]
+    });
+  }
+
+  public async handleInsertAIContent(payload: any): Promise<void> {
+    const {
+      proseMirrorEditorView,
+      setIsLocked,
+      params,
+      language
+    } = this.context!;
+    const configuration = new AIConfiguration(params);
+
+    const { dispatch } = proseMirrorEditorView;
+    let { state } = proseMirrorEditorView;
+
+    setIsLocked(true);
+
+    let startPosition = state.selection?.from ?? state.doc.content.size;
+    let endPosition = state.selection?.to || startPosition;
+
+    const resolved = state.doc.resolve(startPosition, endPosition);
+    const isEmptyParagraph =
+      resolved?.parent?.type?.name === "paragraph" &&
+      resolved?.parent?.content?.size === 0;
+
+    if (isEmptyParagraph) {
+      startPosition--;
+    }
+
+    let alert: Alert | undefined;
+
+    try {
+      await invokeChatCompletions(
+        configuration,
+        payload,
+        (buffer, complete) => {
+          const isEmpty = buffer.length === 0;
+          const isDialog = buffer.startsWith(DIALOG_PREFIX);
+          const couldBeDialog =
+            !complete &&
+            buffer.length < DIALOG_PREFIX.length &&
+            buffer === DIALOG_PREFIX.slice(0, buffer.length);
+
+          if (isDialog) {
+            const content = buffer.slice(DIALOG_PREFIX.length);
+            if (!alert) {
+              alert = this.context!.dialogs.alert({
+                title: "AI Assistant",
+                icon: <AssistantIcon />,
+                severity: "info",
+                content
+              });
+            } else {
+              alert.updateContent(content);
+            }
+            setIsLocked(false);
+            return;
+          }
+
+          if (isEmpty || couldBeDialog) {
+            return;
+          }
+
+          let fragment = (language as MarkdownLanguage).parseMarkdown(
+            buffer.trim()
+          ).content;
+
+          if (fragment.content.length === 1) {
+            fragment = fragment.content[0].content;
```

Comment:
Are we still happy with this hack? I added it in to try to prevent single lines from being inserted as new paragraphs, but I guess it's possible it might have undesired consequences in other circumstances

Reply:
I tested it a bunch over the weekend with various types, it worked better than not having it but does not cover every case